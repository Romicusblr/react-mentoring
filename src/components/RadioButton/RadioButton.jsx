/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';
import style from './RadioButton.module.css';

function RadioButton({ value, name, defaultChecked }) {
  const radioId = `${name}_${value}`;
  return (
    <p className={style.radioButton}>
      <input type="radio" name={name} id={radioId} defaultChecked={defaultChecked} />
      <label htmlFor={radioId}>{value}</label>
    </p>
  );
}

RadioButton.propTypes = {
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  defaultChecked: PropTypes.bool,
};

RadioButton.defaultProps = {
  defaultChecked: false,
};

export default RadioButton;
