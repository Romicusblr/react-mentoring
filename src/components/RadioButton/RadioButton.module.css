.radioButton {
  &>input {
    appearance: none;
    &:checked + label {
      background-color: var(--color-raised);
    }
  }

  &>label {
    padding: .5rem 1.5rem;
    cursor: pointer;
    background-color: var(--color-secondary);
  }

  &:first-of-type>label {
    border-radius: .25rem 0 0 .25rem;
  }

  &:last-of-type>label {
    border-radius: 0 .25rem .25rem 0 ;
  }
}
