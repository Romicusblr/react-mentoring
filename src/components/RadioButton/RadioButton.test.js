import RadioButton from '.';

fdescribe('RadioButton', () => {
  it('should be selectable by class "radioButton"', () => {
    expect(shallow(<RadioButton />).is('.radioButton')).toBe(true);
  });

  it('should match to snapshot', () => {
    const snap = shallow(<RadioButton />);
    expect(snap).toMatchSnapshot();
  });
});
