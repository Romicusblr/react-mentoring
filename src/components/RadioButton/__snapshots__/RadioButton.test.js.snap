// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should match to snapshot 1`] = `
<p
  className="radioButton"
>
  <input
    defaultChecked={false}
    id="radio_undefined"
    name="radio"
    type="radio"
  />
  <label
    htmlFor="radio_undefined"
  />
</p>
`;
